packages/flutter/lib/src/widgets/form.dart

```diff
@@ -618,13 +641,36 @@ class FormFieldState<T> extends State<FormField<T>> with RestorationMixin {
           if (_hasInteractedByUser.value) {
             _validate();
           }
+        case AutovalidateMode.onUnfocus:
         case AutovalidateMode.disabled:
           break;
       }
     }
+
     Form.maybeOf(context)?._register(this);
+
+    if (Form.maybeOf(context)?.widget.autovalidateMode == AutovalidateMode.onUnfocus && !_isAlwaysAutoValidateMode || widget.autovalidateMode == AutovalidateMode.onUnfocus) {
+      return Focus(
+        canRequestFocus: false,
+        skipTraversal: true,
+        onFocusChange: (bool value) {
+          if (!value) {
+            setState(() {
+              _validate();
+            });
+          }
+        },
+        focusNode: _focusNode,
+        child: widget.builder(this),
+      );
+    }
+
     return widget.builder(this);
   }
+
+  FormFieldState<dynamic>? get _formFieldState => Form.maybeOf(context)?._fields.firstWhereOrNull((FormFieldState<dynamic> field) => field == this);
```

Comment:
What is the point of this over just using `this`? This returns null if there is no Form above this field.

Reply:
My idea of using _formFieldState instead of this directly is to ensure that the reference to the associated FormFieldState is safe, especially when there is no Form widget above the form field.

Reply:
Looking at where this is used I see you also have `widget.autovalidateMode == AutovalidateMode.onUnfocus`. Why not just do `widget.autovalidateMode == AutovalidateMode.always` instead of having these two getters? I see that the logic is slightly different (due to returning false if there is no Form above) but I don't understand why that's necessary. If that's intentional, then maybe these getters need slightly different names, or need comments to explain the nuance.

Reply:
Okay, I actually think that makes more sense.